fe/fe-core/src/main/java/com/starrocks/connector/iceberg/ScalarOperatorToIcebergExpr.java

```diff
@@ -132,10 +132,17 @@ private static Type getResultType(String columnName, IcebergContext context) {
         }
 
         private static Type getColumnType(String qualifiedName, IcebergContext context) {
-            String[] paths = qualifiedName.split("\\.");
             Type type = context.getSchema();
-            for (String path : paths) {
-                type = type.asStructType().fieldType(path);
+            Types.StructType structType = type.asStructType();
+            type = structType.fieldType(qualifiedName);
+            if (null != type) {
+                return type;
+            }
+            if (qualifiedName.contains(".")) {
```

Comment:
why does your column name has dot?

Reply:
For example, we have a business wide table with columns named `event_time.string` and `event_time.int`, which are different event tracking logs with the same field name but different data types

Reply:
The reason why I changed this part of the code is to consider that if iceberg originally supported it, then starrocks could also be able to support it

Reply:
please add an sql test case

Reply:
done